include/nuttx/sched.h

```diff
@@ -526,8 +526,7 @@ struct task_group_s
 #ifndef CONFIG_DISABLE_ENVIRON
   /* Environment variables **************************************************/
 
-  size_t     tg_envsize;            /* Size of environment string allocation    */
-  FAR char  *tg_envp;               /* Allocated environment strings            */
+  FAR char **tg_envp;               /* Allocated environment strings            */
```

Comment:
It is fine for now, but we can consider an improvement of runtime speed by adding `size_t tg_envc;` and not do not perform
```
while (ptcb->group->tg_envp[envc] != NULL)
  {
    envc++;
  }
```
every time we need to get the number of variables. Adding it will also eliminate requirement for `group->tg_envp` to be NULL-terminated.

Reply:
> It is fine for now, but we can consider an improvement of runtime speed by adding `size_t tg_envc;` and not do not perform
> 
> ```
> while (ptcb->group->tg_envp[envc] != NULL)
>   {
>     envc++;
>   }
> ```
> 
> every time we need to get the number of variables.

I consider this before, but it require the caller pass envc along envp, which isn't easy with the current api, e.g.:
```
int execve(const char *pathname, char *const argv[], char *const envp[]);
int posix_spawn(pid_t *restrict pid, const char *restrict path,
                           const posix_spawn_file_actions_t *restrict file_actions,
                           const posix_spawnattr_t *restrict attrp,
                           char *const argv[],
                           char *const envp[]);
```

> Adding it will also eliminate requirement for `group->tg_envp` to be NULL-terminated.

tg_envp need be null terminated to implement get_environ_ptr(environ):
https://man7.org/linux/man-pages/man7/environ.7.html